File: SDKs/twilio.com/twilio_bulkexports_v1/1.36.0/python/src/sdk/models/operations/fetchjob.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared
FETCH_JOB_SERVERS = [
	"https://bulkexports.twilio.com",
]


@dataclass
class FetchJobPathParams:
    job_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'JobSid', 'style': 'simple', 'explode': False }})
    

@dataclass
class FetchJobSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class FetchJobRequest:
    server_url: Optional[str] = field(default=None)
    path_params: FetchJobPathParams = field(default=None)
    security: FetchJobSecurity = field(default=None)
    

@dataclass
class FetchJobResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    bulkexports_v1_export_job: Optional[shared.BulkexportsV1ExportJob] = field(default=None)
    

File: SDKs/twilio.com/twilio_bulkexports_v1/1.36.0/typescript/src/sdk/models/operations/fetchjob.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const FETCHJOB_SERVERS = [
	"https://bulkexports.twilio.com",
];



export class FetchJobPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=JobSid" })
  jobSid: string;
}


export class FetchJobSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class FetchJobRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: FetchJobPathParams;

  @Metadata()
  security: FetchJobSecurity;
}


export class FetchJobResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  bulkexportsV1ExportJob?: shared.BulkexportsV1ExportJob;
}
